apps/gdaltindex_lib.cpp

```diff
@@ -165,6 +166,231 @@ static bool PatternMatch(const char *input, const char *pattern)
+
+    argParser->add_description(
+        _("Build a tile index from a list of datasets."));
+
+    argParser->add_epilog(
+        _("For more details, see the full documentation for gdaltindex at\n"
+          "https://gdal.org/programs/gdaltindex.html"));
+
+    argParser->add_argument("-overwrite")
+        .flag()
+        .store_into(psOptions->bOverwrite)
+        .help(_("Overwrite the output tile index file if it already exists."));
+
+    argParser->add_argument("-recursive")
+        .flag()
+        .store_into(psOptions->bRecursive)
+        .help(_("Whether directories specified in <file_or_dir> should be "
+                "explored recursively."));
+
+    argParser->add_argument("-filename_filter")
+        .metavar("<val>")
+        .append()
+        .store_into(psOptions->oSetFilenameFilters)
+        .help(_("Pattern that the filenames contained in directories pointed "
+                "by <file_or_dir> should follow. '*' is a wildcard character "
+                "that matches any number of any characters including none. '?' "
+                "is a wildcard character that matches a single character. "
+                "Comparisons are done in a case insensitive way. Several "
+                "filters may be specified.."));
+
+    argParser->add_argument("-min_pixel_size")
+        .metavar("<val>")
+        .store_into(psOptions->dfMinPixelSize)
+        .help(_("Minimum pixel size in term of geospatial extent per pixel "
+                "(resolution) that a raster should have to be selected. The "
+                "pixel size is evaluated after reprojection of its extent to "
+                "the target SRS defined by -t_srs."));
+
+    argParser->add_argument("-max_pixel_size")
+        .metavar("<val>")
+        .store_into(psOptions->dfMaxPixelSize)
+        .help(_("Maximum pixel size in term of geospatial extent per pixel "
+                "(resolution) that a raster should have to be selected. The "
+                "pixel size is evaluated after reprojection of its extent to "
+                "the target SRS defined by -t_srs."));
+
+    argParser->add_output_format_argument(psOptions->osFormat);
+
+    argParser->add_argument("-tileindex")
+        .metavar("<field_name>")
+        .store_into(psOptions->osLocationField)
+        .help(_("Name of the layer in the tile index file."));
+
+    argParser->add_argument("-write_absolute_path")
+        .flag()
+        .store_into(psOptions->bWriteAbsolutePath)
+        .help(_("Write the absolute path of the raster files in the tile index "
+                "file."));
+
+    argParser->add_argument("-skip_different_projection")
+        .flag()
+        .store_into(psOptions->bSkipDifferentProjection)
+        .help(_(
+            "Only files with the same projection as files already inserted in "
+            "the tile index will be inserted (unless -t_srs is specified)."));
+
+    argParser->add_argument("-t_srs")
+        .metavar("<target_srs>")
```

Comment:
We use srs_def for similar instances:
```suggestion
        .metavar("<srs_def>")
```